inmemorycompiler/src/main/java/org/togetherjava/tjbot/javap/Javap.java

```diff
@@ -0,0 +1,148 @@
+package org.togetherjava.tjbot.javap;
+
+import org.jetbrains.annotations.NotNull;
+
+import javax.tools.DiagnosticListener;
+import javax.tools.JavaFileManager;
+import javax.tools.JavaFileObject;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * A javap-interface that uses reflection to access locked-down classes of the jdk.jdeps module and
+ * keep everything in-memory
+ */
+public final class Javap {
+    // Hide constructor
+    public Javap() {}
+
+    private static final String TEMP_FILE_NAME = "tmp";
+    private static final String ANNOYING_WARNING_MESSAGE =
+            "Warning: File /%s does not contain class %<s";
+
+    // ======== reflection
+
+    // ---- classes
+    private static final Class<?> R_C_JavapFileManager;
+    private static final Class<?> R_C_JavapTask;
+
+    // ---- methods
+    private static final Method R_C_JavapFileManager_M_create;
+    private static final Method R_C_JavapTask_M_run;
+    private static final Method R_C_JavapTask_M_getDiagnosticListenerForWriter;
+
+    // ---- fields
+    private static final Field R_C_JavapTask_F_defaultFileManager;
+    private static final Field R_C_JavapTask_F_log;
+
+    static {
+        try {
+            R_C_JavapFileManager = Class.forName("com.sun.tools.javap.JavapFileManager");
+            R_C_JavapFileManager_M_create = R_C_JavapFileManager.getDeclaredMethod("create",
+                    DiagnosticListener.class, PrintWriter.class);
+
+            R_C_JavapTask = Class.forName("com.sun.tools.javap.JavapTask");
+            R_C_JavapTask_F_defaultFileManager =
+                    R_C_JavapTask.getDeclaredField("defaultFileManager");
+            R_C_JavapTask_F_log = R_C_JavapTask.getDeclaredField("log");
+            R_C_JavapTask_M_run = R_C_JavapTask.getDeclaredMethod("run", String[].class);
+            R_C_JavapTask_M_getDiagnosticListenerForWriter =
+                    R_C_JavapTask.getDeclaredMethod("getDiagnosticListenerForWriter", Writer.class);
+
+            // ------- make things accessible
+
+            R_C_JavapFileManager_M_create.setAccessible(true);
+            R_C_JavapTask_F_defaultFileManager.setAccessible(true);
+            R_C_JavapTask_F_log.setAccessible(true);
+            R_C_JavapTask_M_run.setAccessible(true);
+            R_C_JavapTask_M_getDiagnosticListenerForWriter.setAccessible(true);
+        } catch (ClassNotFoundException | NoSuchMethodException | NoSuchFieldException ex) {
+            throw new ReflectionException(
+                    "A fatal exception has occurred while using reflection in the non-static initializer of org.togetherjava.tjbot.javap.Javap",
+                    ex);
+        }
+    }
+
+    // ========
```

Comment:
How does this get generated, how can I generate it myself? Why do we even have generated code in the first place and dont adjust it to make it "nice and proper"?

This generated code trips me, in particular since it has no comment/explanation/instructions whatsoever.

Reply:
I cleaned up the names and made it overall more appealing to the eyes.

Reply:
So its not intended to be regenerated or kept up to date with the generator tool? You simply used it as a one-time thing and now its "manual code"?

Reply:
Yes.